src/feeds/FeedReader.ts

```diff
@@ -365,40 +364,44 @@ export class FeedReader {
     /**
      * Start polling all the feeds. 
      */
-    public async pollFeeds(): Promise<void> {
-        log.debug(`Checking for updates in ${this.observedFeedUrls.size} RSS/Atom feeds`);
+    public async pollFeeds(workerId: number): Promise<void> {
+
+        // Update on each iteration
+        Metrics.feedsFailing.set({ reason: "http" }, this.feedsFailingHttp.size );
+        Metrics.feedsFailing.set({ reason: "parsing" }, this.feedsFailingParsing.size);
+        Metrics.feedsFailingDeprecated.set({ reason: "http" }, this.feedsFailingHttp.size );
+        Metrics.feedsFailingDeprecated.set({ reason: "parsing" }, this.feedsFailingParsing.size);
```

Comment:
Since `this.pollFeed` is what updates these counts, does updating the metrics before the awaited call to `this.pollFeed` mean there's a risk of each worker making overlapping updates of the same values?

Should there be an async lock on making these updates?

Reply:
It should be fine. This section of code just checks the size of a Set and updates the metric each time any worker finishes, overlapping updates are fine because the workers are all updating the same shared value, so as long as it gets updated each time a worker finishes it will be correct.

Reply:
It looks like they get updated when each worker _starts_, not finishes; but that should be fine since the workers run periodically anyways.